Apps/Src/ReadCarCAN.c

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2020 UT Longhorn Racing Solar */
+
+#include "ReadCarCAN.h"
+
+#define THRESHOLD 1
+
+void Task_ReadCarCAN(void *p_arg) {
+    car_state_t *car = (car_state_t *) p_arg;
+
+    uint8_t buffer[8]; // buffer for CAN message
+
+    static int faultCounter = 0;
+
+    OS_ERR err;
+    CPU_TS ts;
+
+    while (1) {
+        OSTaskSemPend(0, OS_OPT_PEND_NON_BLOCKING, &ts, &err);
+        if (err == OS_ERR_NONE) {
+            // A signal was received, so the task should wait until signaled again
+            OSTaskSemPend(0, OS_OPT_PEND_BLOCKING, &ts, &err);
+            // TODO: error handling
+        } else if (err != OS_ERR_PEND_WOULD_BLOCK) {
+            // TODO: error handling
+        }
+
+        // Normal task countinues here
+
+        // Check if BPS told us to shut down the motor at least THRESHOLD times
+        if (CANbus_Read(buffer) == SUCCESS && (++faultCounter == THRESHOLD)) {
```

Comment:
Also need to check what the actual message is. BPS will send either a 0 (charging disabled) or a 1 (charging enabled). If charging is disabled, we need to disconnect the contactors. If BPS is sending a 1, we don't need to change anything about the connections

Reply:
Also, the threshold isn't really needed here. You should use the threshold to check if the message was received at all (i.e.) if `CANbus_Read` returns ERROR, that means BPS hasn't sent the message. If that happens THRESHOLD times, we should disconnect.